src/ResourceManager/Resources/Commands.Resources/Models.ResourceGroups/ResourceIdentifier.cs

```diff
@@ -138,10 +138,18 @@ public override string ToString()
 
             AppendIfNotNull(ref resourceId, "/subscriptions/{0}", Subscription);
             AppendIfNotNull(ref resourceId, "/resourceGroups/{0}", ResourceGroupName);
-            AppendIfNotNull(ref resourceId, "/providers/{0}", provider);
-            AppendIfNotNull(ref resourceId, "/{0}", parentAndType);
-            AppendIfNotNull(ref resourceId, "/{0}", ResourceName);
 
+            // Only append parentAndType and ResourceName when provider is not empty.
+            if (!string.IsNullOrEmpty(provider))
+            {
+                resourceId.AppendFormat("/providers/{0}", provider);
+                if (!string.IsNullOrEmpty(parentAndType))
+                {
+                    resourceId.AppendFormat("/{0}", parentAndType);
+                    AppendIfNotNull(ref resourceId, "/{0}", ResourceName);
+                }
+            }
```

Comment:
this change might end up silently absorbing invalid inputs. i liked the option of not changing anything here and validating the final scope better